frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1938,18 +1939,33 @@ end
 function ReaderHighlight:onMoveHighlightIndicator(args)
     if self.view.visible_area and self._current_indicator_pos then
         local dx, dy, quick_move = unpack(args)
-        local step_distance = self.view.visible_area.w / 5 -- quick move distance: fifth of visible_area
-        local y_step_distance = self.view.visible_area.h / 5
-        if step_distance > y_step_distance then
-            -- take the smaller, make all direction move distance much predictable
-            step_distance = y_step_distance
-        end
-        if not quick_move then
-            step_distance = step_distance / 4 -- twentieth of visible_area
-        end
+        local quickly_move_distance_dx = self.view.visible_area.w / 5 -- quick move distance: fifth of visible_area
+        local quickly_move_distance_dy = self.view.visible_area.h / 5
+        -- single move distance, small and capable to move on word with small font size and narrow line height
+        local move_distance = Size.item.height_default / 4
         local rect = self._current_indicator_pos:copy()
-        rect.x = rect.x + step_distance * dx
-        rect.y = rect.y + step_distance * dy
+        if quick_move then
+            rect.x = rect.x + quickly_move_distance_dx * dx
+            rect.y = rect.y + quickly_move_distance_dy * dy
+        else
+            local now = TimeVal:now()
+            if dx == self._last_indicator_move_args.dx and dy == self._last_indicator_move_args.dy then
+                local diff = now - self._last_indicator_move_args.time
+                -- if press same arrow key in 1 second, speed up
+                -- double press: 4 single move distances, usually move to next word or line
+                -- triple press: 16 single distances, usually skip several words or lines
+                -- quadruple press: 54 single distances, almost move to screen edge
+                if diff < TimeVal:new{ sec = 1, usec = 0 } then
+                    move_distance = self._last_indicator_move_args.distance * 4
```

Comment:
No need to somehow check and limit this to 4 (so you don't get into quintuple :)? Or just the dexterity it would take to do 5 x press it itself limiting ?

Reply:
No need to. I tested. If 5 or more quick press, the move distance will hit edge guard and the indicator will stay at screen edge.
https://github.com/koreader/koreader/blob/c9bfb74d2aa9d9cbdffbcc6cffc3f79ac5f1eec0/frontend/apps/reader/modules/readerhighlight.lua#L1953-L1964

Non touch devices have limited performance, it takes time to run event handle code and repaint changes. It still takes sensible time to response in my Kindle DX even though small piece of screen region need repainted.

For my experience, press 3 or 4 times array key, wait for a second to see where the indicator at. Do another further(3x) move or small(2x) move make indicator close the target position.

Do 5 or more press is unless unless want move indicator to edge.